aybapi/base/client.py

```diff
@@ -0,0 +1,36 @@
+import abc
+import typing
+from ..impl.bot import Bot
+from ..impl.category import Category
+
+
+class Client(abc.ABC):
+
+    @abc.abstractmethod
+    def fetch_stats(self) -> typing.Optional[dict]:
+        """Fetches the overall site status available to the public
+
+        Returns
+        -------
+        API response: typing.Dict
+                The API response containing the site status information.
+        """
+
+    @abc.abstractmethod
+    def fetch_bot(self, id) -> Bot:
+        """Fetches a specific bot according to it's ID
+
+        Returns
+        -------
+        bot: aybapi.Bot
+                AYBAPI bot object containing bot information from API response.
+        """
+
+    def fetch_category(self, id) -> Category:
```

Comment:
Why would you ever need more than one implementation for this? This feels like massively overengineering the problem you are trying to solve.

It isn't like this would even be compatible between async and sync clients anyway, since coroutines would have to return `typing.Coroutine[typing.Any, typing.Any, Category]` instead of `Category`.